recipes/b2/portable/conanfile.py

```diff
@@ -53,49 +56,89 @@ def validate(self):
             raise ConanInvalidConfiguration(
                 "Option toolset 'cxx' and 'cross-cxx' requires 'use_cxx_env=True'")
 
+    def layout(self):
+        conan.tools.layout.basic_layout(self, src_folder="source")
```

Comment:
For consistency, use `src` as the folder name.

```suggestion
        conan.tools.layout.basic_layout(self, src_folder="src")
```

Reply:
Consistent with the templates that we are using on ConanCenterIndex: https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/conanfile.py

Please, try keeping the same pattern.

Reply:
@uilianries I don't see `basic_layout` in https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/conanfile.py, what exactly do you refer (which line)?
anyway, let's update either docs or package_templates, to be consistent with each other, otherwise it's too confusing

Reply:
@SSE4 In that case, autotools template is better: https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/autotools_package/all/conanfile.py

It uses basic_setup. Thank you for pointing!

Reply:
@uilianries okay, good. let's update our `package_templates` to match the migration guide.
because right now, people read migration guide, follow it, upgrade their recipes, and then, during the review, you tell them their changes are wrong and they have to start from scratch (for consistency or whatever) - it's not nice, let's avoid double work.

Reply:
Are there any recipes where the length of the `src_folder` name might cause path length issues on Windows? We'd have to update a lot of recipes on CCI already to use `src`. Might be worth updating the migration guide?

@grafikrobot I'm sure that once we come to a conclusion we'll let you know. For now, you probably don't need to worry about. I don't think the source folder name will block your PR. We can always revisit it with a follow up tweak if necessary. Thanks for being patient with us!

Reply:
Reading the comments, everyone agrees about the function benefits and style benefits. 

It important to remember that CCI is much more specific then the Conan docs which is why the difference exists.
Practically speaking the `src` is a much more common name used by projects to put the `.cpp` files which is why it's used here. But that really does not hold much meaning since grab the tarbells and not inline recipes...

Given there are no approvals and the build service is not drowning I would encourage you to apply the suggestion no about no technical reason.

This is by no means a blocker, reviewers are not required to approve that why we are a community so different opinions can lead to the best quality.

Given I like 💄 on my 🐷  please apply suggestion but I will gladly ignore this because this is a requirement for boost 😄